src/native-common/utils/EventHelpers.ts

```diff
@@ -0,0 +1,144 @@
+* Copyright (c) Microsoft Corporation. All rights reserved.
+* Licensed under the MIT license.
+*/
+import _ = require('../../native-common/lodashMini');
+import Types = require('../../common/Types');
+
+//
+// These helpers promote a SyntheticEvent to their higher level counterparts
+export class EventHelpers {
+
+    toKeyboardEvent(e: Types.SyntheticEvent): Types.KeyboardEvent {
+        // Conversion to a KeyboardEvent-like event if needed
+        let keyEvent = e as Types.KeyboardEvent;
+        if (keyEvent.keyCode === undefined) {
+
+            // Currently some key codes are dependent on platform. React Native proper (the iOS and Android platforms) have different
+            // keycodes for arrow keys when comparing with React (JS).
+            // We align the keycodes for native desktop platforms to the other native ones, as a workaround.
+            // Ideally all key codes should be consistent OR a set of constants should be exposed by ReactXP.
+            let keyName: string = (e.nativeEvent as any).key;
+            let keyCode: number = 0;
+
+            if (keyName.length === 1) {
+                keyCode = keyName.charCodeAt(0);
+            } else {
+                switch (keyName) {
+                    case 'Backspace':
+                        keyCode = 8;
+                        break;
+
+                    case 'Tab':
+                        keyCode = 9;
+                        break;
+
+                    case 'Enter':
+                        keyCode = 13;
+                        break;
+
+                    case 'Shift':
+                        keyCode = 16;
+                        break;
+
+                    case 'Control':
+                        keyCode = 17;
+                        break;
+
+                    case 'Alt':
+                        keyCode = 18;
+                        break;
+
+                    case 'Escape':
+                        keyCode = 27;
+                        break;
+
+                    case 'Space':
+                        keyCode = 32;
+                        break;
+
+                    case 'PageUp':
+                        keyCode = 92;
+                        break;
+
+                    case 'PageDown':
+                        keyCode = 93;
+                        break;
+
+                    case 'Left':
+                    case 'ArrowLeft':
+                        keyCode = 21;
+                        break;
+
+                    case 'Up':
+                    case 'ArrowUp':
+                        keyCode = 19;
+                        break;
+
+                    case 'Right':
+                    case 'ArrowRight':
+                        keyCode = 22;
+                        break;
+
+                    case 'Down':
+                    case 'ArrowDown':
+                        keyCode = 20;
+                        break;
+
+                    case 'Delete':
+                        keyCode = 8;
+                        break;
```

Comment:
Should we move this to be with "Backspace" above which has the same key code?

Reply:
Ok, will do

---
In reply to: [159335721](https://github.com/Microsoft/reactxp/pull/404#discussion_r159335721) [](ancestors = 159335721)